### PR TITLE
Bug 1170125 - [NGA][Contacts] Extract everything related to "mozConta…

### PR DESCRIPTION
**Pull request by borjasalguero**
…cts" to a separate .js file r=arcturus

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=38c173977cf2c26c5c8775147fee1fc828cf05b8

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9a0ec1428cc342b4252cfd8bfcdd075e13cdce09

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1465d46577948a24fcea53612fb35d5bbb865711

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bcf4efd3bda0dfe00d27441cb220fc265896f3e0

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0c6e141c987795e47c87dfd761ec4f3f344a3801

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=53e8ddffaaa9901d1432c3a8f308a253907c6faa

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=53e8ddffaaa9901d1432c3a8f308a253907c6faa

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=85392ee9b4eba860d7f69159bfffc56e69701c8e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f8296ac7ea8ad823d14acd297323ffa22724a305

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ace8c4bd01b1787571660faf8a62aa95ca1b5d93

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=81c9f75dc99c100436fcf6ae30ca79f6b16a7955

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7cabdc672b8924fd86d18673f1054ef72242fc5a

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f7e000b4cadf2c0818de9643ae26297ec3c9bef3

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=58c4073a9e6cab9b56ac1ac276d5f06ce60af0d5

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a684c4a981d0acdf21b97e97ba9763798478ce11

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f5229afcdcc95a5ada8ea47b429da3b8797dc76f

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1f18ada42991a9469cc7a95c4cc411b44d928b97

File: apps/communications/contacts/js/views/settings.js [hunk 1]
```diff
@@ -988,27 +989,28 @@ contacts.Settings = (function() {
 
   var checkNoContacts = function checkNoContacts() {
     var exportButton = exportContacts.firstElementChild;
-    var req = navigator.mozContacts.getCount();
-    req.onsuccess = function() {
-      if (req.result === 0) {
+
+    ContactsService.isDBEmpty(function(error, isEmpty) {
```
Comment (arcturus) [inline apps/communications/contacts/js/views/settings.js]: nit: rename the method to `isEmpty`

File: apps/communications/contacts/services/contacts.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+   */
+
+
+  // All events based on mozConcacts will be handled by this var
+  var events = {};
+
+  // Execute all handlers based on an event type
+  function executeHandlers(e) {
+    var handlers = events[e.type];
+    for (var i = 0; i < handlers.length; i++) {
```
Comment (arcturus) [inline apps/communications/contacts/services/contacts.js]: handlers could pottentially be null and `handlers.length` throw an error.

**Comment by arcturus** [comment]
There is a call to mozContacts.find() to fetch all contacts that can be removed from list.js:

https://patch-diff.githubusercontent.com/raw/mozilla-b2g/gaia/pull/30397.diff

File: apps/communications/contacts/js/views/settings.js (hunk 1 above)
Comment (borjasalguero) [inline apps/communications/contacts/js/views/settings.js]: +1

File: apps/communications/contacts/services/contacts.js (hunk 2 above)
Comment (borjasalguero) [inline apps/communications/contacts/services/contacts.js]: +1

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=09d4d97502b8a49ae2d03a6b09157decc915eec2

**Comment by borjasalguero** [comment]
All comments addressed. Waiting to have all tests passed to merge!

**Pull request merged**